src/app/TimedRequest.cpp

```diff
@@ -33,10 +33,10 @@ using namespace Messaging;
 CHIP_ERROR TimedRequest::Send(ExchangeContext * aExchangeContext, uint16_t aTimeoutMs)
 {
     // The payload is an anonymous struct (2 bytes) containing a single
-    // 16-bit integer with a context tag (1 control byte, 1 byte tag, at
-    // most 2 bytes for the integer).  Use MessagePacketBuffer::New to
-    // account for other message-global overheads (MIC, etc).
-    System::PacketBufferHandle payload = MessagePacketBuffer::New(6);
+    // 16-bit integer with two context tag (1 control byte, 1 byte tag, at
+    // most 2 bytes for the timeout integer and 1 control byte, 1 byte tag, one byte for InteractionModelRevision).  Use
+    // MessagePacketBuffer::New to account for other message-global overheads (MIC, etc).
+    System::PacketBufferHandle payload = MessagePacketBuffer::New(9);
```

Comment:
So, I don't think that the revision tag should actually be encoded right at the end. That results in all kinds of fix-ups to these heuristics, which are not necessary.

Reply:
Currently only TimedRequest have this issue, I think it is fine.